seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/catalog/HbaseCatalog.java

```diff
@@ -0,0 +1,206 @@
+import org.apache.seatunnel.api.table.catalog.exception.TableAlreadyExistException;
+import org.apache.seatunnel.api.table.catalog.exception.TableNotExistException;
+import org.apache.seatunnel.connectors.seatunnel.hbase.client.HbaseClient;
+import org.apache.seatunnel.connectors.seatunnel.hbase.config.HbaseParameters;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/** Hbase catalog implementation. */
+@Slf4j
+public class HbaseCatalog implements Catalog {
+
+    private final String catalogName;
+    private final String defaultDatabase;
+    private final HbaseParameters hbaseParameters;
+
+    private HbaseClient hbaseClient;
+
+    public HbaseCatalog(
+            String catalogName, String defaultDatabase, HbaseParameters hbaseParameters) {
+        this.catalogName = checkNotNull(catalogName, "catalogName cannot be null");
+        this.defaultDatabase = defaultDatabase;
+        this.hbaseParameters = checkNotNull(hbaseParameters, "Hbase Config cannot be null");
+    }
+
+    @Override
+    public void open() throws CatalogException {
+        try {
+            hbaseClient = HbaseClient.createInstance(hbaseParameters);
+        } catch (Exception e) {
+            throw new CatalogException(String.format("Failed to open catalog %s", catalogName), e);
+        }
+    }
+
+    @Override
+    public void close() throws CatalogException {
+        hbaseClient.close();
+    }
+
+    @Override
+    public String name() {
+        return catalogName;
+    }
+
+    @Override
+    public String getDefaultDatabase() throws CatalogException {
+        return defaultDatabase;
+    }
+
+    @Override
+    public boolean databaseExists(String databaseName) throws CatalogException {
+        return hbaseClient.databaseExists(databaseName);
+    }
+
+    @Override
+    public List<String> listDatabases() throws CatalogException {
+        return hbaseClient.listDatabases();
+    }
+
+    @Override
+    public List<String> listTables(String databaseName)
+            throws CatalogException, DatabaseNotExistException {
+        if (!databaseExists(databaseName)) {
+            throw new DatabaseNotExistException(catalogName, databaseName);
+        }
+        return hbaseClient.listTables(databaseName);
+    }
+
+    @Override
+    public boolean tableExists(TablePath tablePath) throws CatalogException {
+        checkNotNull(tablePath);
+        return hbaseClient.tableExists(tablePath.getTableName());
+    }
+
+    @Override
+    public CatalogTable getTable(TablePath tablePath)
+            throws CatalogException, TableNotExistException {
+        checkNotNull(tablePath, "tablePath cannot be null");
+        // Hbase cannot retrieve the columns of the table
+        return CatalogTable.of(
```

Comment:
why not throw unsupportedexception? The catalogtable returned by this method is useless.

Reply:
Already modified